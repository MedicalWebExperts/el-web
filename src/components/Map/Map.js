import React from 'react';
import { string, shape } from 'prop-types';
import css from 'styled-jsx/css';

const propTypes = {
  title: string,
  url: string.isRequired,
  styles: shape({}),
};

const defaultProps = {
  title: 'Map',
  styles: {},
};

const customStyle = css`
  border: 0;
`;

const style = { height: '100%', width: '100%' };

const Map = ({ title, url, styles }) => (
  <iframe
    title={title}
    src={url}
    height={style.height}
    width={style.width}
    frameBorder="0"
    allowFullScreen
    style={styles}
  >
    <style jsx>{customStyle}</style>
  </iframe>
);

Map.propTypes = propTypes;
Map.defaultProps = defaultProps;
export default Map;
