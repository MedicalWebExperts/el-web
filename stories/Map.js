import React from 'react';
import { storiesOf } from '@storybook/react';

import { Map } from '../src';

export default () => {
  storiesOf('Map', module).add('Default', () => (
    <Map
      url="https://www.google.com/maps/embed?pb=!1m14!1m8!1m3!1d13138.598992769137!2d-58.430499!3d-34.587728!3m2!1i1024!2i768!4f13.1!3m3!1m2!1s0x0%3A0x910a40368a2b8c66!2sChori!5e0!3m2!1sen!2sar!4v1536682900333"
      styles={{ height: '600px' }}
    />
  ));
};
