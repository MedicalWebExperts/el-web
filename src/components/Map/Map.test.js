import React from 'react';
import renderer from 'react-test-renderer';
import Map from './Map';

const defaultProps = {
  url:
    'https://www.google.com/maps/embed?pb=!1m14!1m8!1m3!1d13138.598992769137!2d-58.430499!3d-34.587728!3m2!1i1024!2i768!4f13.1!3m3!1m2!1s0x0%3A0x910a40368a2b8c66!2sChori!5e0!3m2!1sen!2sar!4v1536682900333',
};

const customProps = {
  title: 'Map test',
};

const customStyles = {
  border: '1px solid red',
};

describe('Component Map Snapshot', () => {
  it('without arguments, take the default values', () => {
    const tree = renderer.create(<Map {...defaultProps} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('diferent iframe title', () => {
    const tree = renderer
      .create(<Map {...Object.assign({}, defaultProps, customProps)} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('using custom styles', () => {
    const tree = renderer.create(<Map {...defaultProps} styles={customStyles} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
