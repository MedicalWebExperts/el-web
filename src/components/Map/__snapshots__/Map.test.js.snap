// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Map Snapshot diferent iframe title 1`] = `
<iframe
  allowFullScreen={true}
  className="jsx-1631902045"
  frameBorder="0"
  height="100%"
  src="https://www.google.com/maps/embed?pb=!1m14!1m8!1m3!1d13138.598992769137!2d-58.430499!3d-34.587728!3m2!1i1024!2i768!4f13.1!3m3!1m2!1s0x0%3A0x910a40368a2b8c66!2sChori!5e0!3m2!1sen!2sar!4v1536682900333"
  style={Object {}}
  title="Map test"
  width="100%"
/>
`;

exports[`Component Map Snapshot using custom styles 1`] = `
<iframe
  allowFullScreen={true}
  className="jsx-1631902045"
  frameBorder="0"
  height="100%"
  src="https://www.google.com/maps/embed?pb=!1m14!1m8!1m3!1d13138.598992769137!2d-58.430499!3d-34.587728!3m2!1i1024!2i768!4f13.1!3m3!1m2!1s0x0%3A0x910a40368a2b8c66!2sChori!5e0!3m2!1sen!2sar!4v1536682900333"
  style={
    Object {
      "border": "1px solid red",
    }
  }
  title="Map"
  width="100%"
/>
`;

exports[`Component Map Snapshot without arguments, take the default values 1`] = `
<iframe
  allowFullScreen={true}
  className="jsx-1631902045"
  frameBorder="0"
  height="100%"
  src="https://www.google.com/maps/embed?pb=!1m14!1m8!1m3!1d13138.598992769137!2d-58.430499!3d-34.587728!3m2!1i1024!2i768!4f13.1!3m3!1m2!1s0x0%3A0x910a40368a2b8c66!2sChori!5e0!3m2!1sen!2sar!4v1536682900333"
  style={Object {}}
  title="Map"
  width="100%"
/>
`;
